snapcraft/internal/errors.py

```diff
@@ -42,15 +39,35 @@ def get_exit_code(self):
 
 
 class SnapcraftReportableError(SnapcraftError):
-    """Base class for all snapcraft exceptions that integrate with Sentry.
+    """DEPRECATED: Use SnapcraftException instead with reportable=True."""
 
-    An exception class inheriting from this will get a prompt to report
-    to sentry if raised and is handled by the exception handler on program
-    exit.
 
-    :cvar fmt: A format string that daughter classes override
+class SnapcraftException(Exception, ABC):
```

Comment:
if ABC isn't really working here due to inheriting from Exception, then raising NotImplementedError instead of wrapping with @abstractmethod might be better.

In any case, doesn't this need to be
```python
class SnapcraftException(Exception, meta=ABC):
```
?

Reply:
ABC does work here, with mypy detecting the issue:
`error: Cannot instantiate abstract class 'TestSnapcraftException' with abstract attribute 'get_brief'`

WRT meta=ABC, I think you may be thinking of the other option to use `metaclass=ABCMeta`.  ABC is a helper class which does this for you: 
https://docs.python.org/3/library/abc.html